### PR TITLE
[UC18#152] Added functionality for users to delete their own custom decks

### LINKED ISSUE #152: Contattare RPC removeCustomDeck e rimuovere il mazzo dalla pagina dei deck [UC18]
_**Dod**_
- [x]  La pull request ha un titolo e descrizione adeguata
- [x]  La pull request è stata collegata con la issue del task
- [x]  I requisiti funzionali del task collegato sono soddisfatti
- [x]  Il codice si compila senza errori
- [x]  Gli Unit Tests sono stati scritti e passati 
- [x]  Gli Unit Tests esistenti sono stati passati
- [x]  Il Code Review è stato fatto
- [ ]  La documentazione è stata aggiornata
- [x]  La pull request è stata accettata e fatto merge sul branch development.

 **Effort effettivo:**

### PR DESCRIPTION
**Pull request by alessioarcara**
This PR implements #152 and adds a new method called deleteCustomDeck in the DecksActivity to remove a custom deck via RPC call. Once the custom deck is successfully deleted, it is removed from the list of decks on the page.
A function was passed to DeckWidget instead of a button to get reference to the DeckWidget.
Tests for the new method were also added.

### CODE REVIEW

File: src/main/java/com/aadm/cardexchange/client/widgets/DeckWidget.java [hunk 1]
```diff
@@ -36,7 +35,7 @@ public class DeckWidget extends Composite implements ImperativeHandleCardSelecti
     HTMLPanel actions;
 
     @UiConstructor
-    public DeckWidget(ImperativeHandleDeck deckHandler, Button removeButton, ImperativeHandleCardsSelection selectionHandler, String name) {
+    public DeckWidget(ImperativeHandleDeck deckHandler, ImperativeHandleDeckRemove removeHandler, ImperativeHandleCardsSelection selectionHandler, String name) {
```
Comment (Matte0Sacc0) [inline src/main/java/com/aadm/cardexchange/client/widgets/DeckWidget.java:38]: Glad to see you've followed the suggestion I've given in the past reviews of naming the ImperativeHandlers with a more specific typo then "parent1, parent2, ..."

File: src/main/java/com/aadm/cardexchange/client/widgets/DeckWidget.java [hunk 2]
```diff
@@ -54,7 +53,20 @@ public DeckWidget(ImperativeHandleDeck deckHandler, Button removeButton, Imperat
...
 
-        if (removeButton != null) actions.add(removeButton);
+        if (removeHandler != null) {
+            Button removeButton = new Button("x", (ClickHandler) e -> {
+                if (Window.confirm("Are you sure you want to remove this deck?"))
+                    removeHandler.onClickRemoveCustomDeck(deckName.getInnerText(), (Boolean isRemoved) -> {
+                        if (isRemoved) {
+                            removeFromParent();
+                        } else {
+                            Window.alert("Deck cannot be deleted. It may be a default deck or does not exist.");
```
Comment (Matte0Sacc0) [inline src/main/java/com/aadm/cardexchange/client/widgets/DeckWidget.java:63]: We could think about a good refactor for this method to return more specific information about the errors in the future, but it's not up to you and this task

**Review by Matte0Sacc0 — APPROVED** [review]
Nothing to say, consistent and well executed!!!
I'm appreciating you've shrinked down your PR that much, they're a lot more manageable!!!

**Review by alessiacrimaldi — APPROVED** [review]
Allright, great job Alessio!

**Pull request merged**